README.md

```diff
@@ -235,6 +235,41 @@ to run. Since arguments are specified in the user interface via text area
 inputs, it's important to check that they conform to the format your Task
 expects, and to sanitize any inputs if necessary.
 
+### Using Task Callbacks
+
+The Task provides callbacks that hook into its life cycle. Available callbacks are:
+
+`on_start`
+`on_pause`
+`on_interrupt`
+`on_error`
+`on_cancel`
+`on_complete`
+
+```ruby
+module Maintenance
+  class UpdatePostsTask < MaintenanceTasks::Task
+    on_start :notify
+
+    def notify
+      NotifyJob.perform_later(self.class.name)
+    end
+
+    # ...
+  end
+end
+```
+
+To share callback behaviour across all Tasks, you can use a module. If you choose to include the module on `MaintenanceTasks::Task` in an initializer, you should use `on_load`
+
+```ruby
+Rails.autoloaders.main.on_load("MaintenanceTasks::Task") do
```

Comment:
This is lazier than to_prepare though. Any to_prepare hook will run on each boot which could slow down an app. Since the Zeitwerk autoloader is default in Rails now we can assume most people will have it.

Reply:
> This is lazier than to_prepare though. 

@gmcgibbon I'm assuming you meant lazier than `on_load` (but please correct me if I've misinterpreted)

Do you recommend instead the initial suggestion then?

```ruby
Rails.autoloaders.main.on_load("MaintenanceTasks::Task") do
  MaintenanceTasks::Task.after_start :notify
end
```